src/dex/aave-v3-stata/aave-v3-stata.ts

```diff
@@ -0,0 +1,507 @@
+  //     srcToken,
+  //     srcAmount,
+  //     destToken,
+  //     destAmount,
+  //     swapData,
+  //     exchange,
+  //   );
+  // }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: AaveV3StataData,
+    side: SwapSide,
+    _: Context,
+    executorAddress: Address,
+  ): DexExchangeParam {
+    const { exchange, srcType, destType } = data;
+    let swapData;
+    let returnAmountPos = undefined;
+
+    if (side === SwapSide.SELL) {
+      if (srcType === TokenType.STATA_TOKEN) {
+        // e.g. sell srcAmount 100 srcToken stataUSDC for destToken USDC
+        swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.redeem, [
+          srcAmount,
+          recipient, // receiver
+          executorAddress, // owner
+          destType === TokenType.UNDERLYING, // withdraw from aToken
+        ]);
+
+        returnAmountPos = extractReturnAmountPosition(
+          AaveV3Stata.stata,
+          StataFunctions.redeem,
+          '',
+          1,
+        );
+      } else {
+        // sell srcAmount 100 srcToken USDC for destToken stataUSDC
+        swapData = AaveV3Stata.stata.encodeFunctionData(
+          StataFunctions.deposit,
+          [
+            srcAmount,
+            recipient, // receiver
+            0, // referrer (noop)
+            srcType === TokenType.UNDERLYING, // deposit to aave
+          ],
+        );
+        returnAmountPos = extractReturnAmountPosition(
+          AaveV3Stata.stata,
+          StataFunctions.deposit,
+        );
+      }
+    } else {
+      if (srcType === TokenType.STATA_TOKEN) {
+        // e.g. buy destAmount 100 destToken USDC for srcToken stataUSDC
+        swapData = AaveV3Stata.stata.encodeFunctionData(
+          StataFunctions.withdraw,
+          [
+            destAmount,
+            recipient, // receiver
+            executorAddress, // owner
+          ],
+        );
+      } else {
+        // e.g. buy destAmount 100 destToken stataUSDC for srcToken USDC
+        swapData = AaveV3Stata.stata.encodeFunctionData(StataFunctions.mint, [
+          destAmount,
+          recipient,
+        ]);
+      }
+    }
+
+    return {
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: true,
+      exchangeData: swapData,
+      targetExchange: exchange,
+      returnAmountPos,
+    };
+  }
+
+  // This is called once before getTopPoolsForToken is
+  // called for multiple tokens. This can be helpful to
+  // update common state required for calculating
+  // getTopPoolsForToken. It is optional for a DEX
+  // to implement this
+  async updatePoolState(): Promise<void> {
+    // TODO: complete me!
+  }
+
+  // Returns list of top pools based on liquidity. Max
+  // limit number pools should be returned.
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    //TODO: complete me!
```

Comment:
Would be nice to know if there's need for such connectors. I assume we might miss some routes if new tokens (stata) will be available on other dexes like Uniswap

Reply:
how exactly does this method work?

Currently most liquidity is in balancer v2 e-cpl rehype pools. Anyhow would be good to have this merged soonish to have basic support and improve afterwards. Happy to help with that.